[AArch64] Support commuted operands in performFlagSettingCombine

c-rhodes opened the pull request:
Fixes #122624.

Assisted-by: gpt-5-codex

c-rhodes committed:
[AArch64] Support commuted operands in performFlagSettingCombine

Fixes #122624.

Assisted-by: gpt-5-codex

MacDue commented on llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:
```diff
@@ -26193,9 +26193,21 @@ static SDValue performFlagSettingCombine(SDNode *N,
-  if (SDNode *Generic = DCI.DAG.getNodeIfExists(
-          GenericOpcode, DCI.DAG.getVTList(VT), {LHS, RHS}))
-    DCI.CombineTo(Generic, SDValue(N, 0));
+  auto CombineWithExistingGeneric = [&](SDValue Op0, SDValue Op1) {
```
nit: I found "`With`" a little confusing here (as the generic operation is not being combined). 
```suggestion
  auto CombineToExistingGeneric = [&](SDValue Op0, SDValue Op1) {
```

Also, the comment above would be outdated in the commuted operands case (as the existing generic node is not _identical_, but equivalent) .

c-rhodes committed:
address comments

MacDue reviewed (approved):
LGTM, thanks :+1: I think Paul's suggestion is a good follow-up.

c-rhodes committed:
address comments

c-rhodes committed:
address comments

- move Glue check outside of lambda
- drop SDLoc() from FindNodeOrInsertPos call
